### ISSUE TITLE
torch.nn.functional.grid_sample outputs NaN

### ISSUE THREAD

**Issue by DNXie** [pytorch/pytorch#51911]
## 🐛 Bug

`torch.nn.functional.grid_sample` outputs NaN if `grid` contains large value

## To Reproduce
~~~python
import torch
torch.nn.functional.grid_sample(input=torch.ones([1,1,1,5]), grid=torch.tensor([[[[ 2.9839e+38, -3.2406e+38]]]]))
~~~

Output:
~~~python
tensor([[[[nan]]]])
~~~
## Expected behavior

Expect a grace exception message if the parameters are invalid or unexpected

## Environment
PyTorch version: 1.8.0.dev20210126+cpu
Is debug build: False
CUDA used to build PyTorch: None
ROCM used to build PyTorch: N/A

OS: Ubuntu 18.04.4 LTS (x86_64)
GCC version: Could not collect
Clang version: Could not collect
CMake version: Could not collect

Python version: 3.7 (64-bit runtime)
Is CUDA available: False
CUDA runtime version: No CUDA
GPU models and configuration: No CUDA
Nvidia driver version: No CUDA
cuDNN version: No CUDA
HIP runtime version: N/A
MIOpen runtime version: N/A

**Comment by mruberry** [comment]
I think this is expected. We typically don't (or can't) raise exceptions based on tensor values (although we do in some cases on the CPU).

**Comment by albanD** [comment]
@mruberry from the doc, any value value of the grid outside [-1, 1] should return the padding. In this case, it is zero padding so it sounds ok to expect a 0 for this kind of grid?

**Comment by mruberry** [comment]
> @mruberry from the doc, any value value of the grid outside [-1, 1] should return the padding. In this case, it is zero padding so it sounds ok to expect a 0 for this kind of grid?

Could be.
